src/main/java/com/diffmin/patch/PatchGeneration.java

```diff
@@ -167,11 +168,11 @@ public int getSrcNodeIndex(CtElement srcNode) {
     }
 
     private static CtElement getMappingOfParent(CtElement insertedNode, SpoonMapping mapping) {
-        CtElement parentElementInPrevModel = mapping.get(insertedNode.getParent());
-        if (parentElementInPrevModel == null) {
+        Optional<CtElement> parentElementInPrevModel = mapping.get(insertedNode.getParent());
+        if (parentElementInPrevModel.isEmpty()) {
             return insertedNode.getParent();
         }
-        return parentElementInPrevModel;
+        return parentElementInPrevModel.get();
```

Comment:
You can rewrite this whole method like this:

```java
return mapping.get(insertedNode.getParent()).orElseGet(insertedNode::getParent);
```

Arguably, the method itself isn't even worth keeping anymore as it's only used in one place, and can be captured with a single expression.